docs/advanced/automated-testing.md

```diff
@@ -0,0 +1,262 @@
+---
+title: Unit Testing
+---
+
+## Getting started
+
+This guide is going to explore unit testing, it’s assuming some familiarity with the subject, but it’s not required.
+
+So to start off, you’re gonna need some project set up, the specifics don’t really matter, you can just take these ideas and extrapolate them to your own needs, but for demonstration I’m going to have a basic project set up like this:
+
+![Project setup](/images/testing-project-setup.png)
+
+Our code is very simple for this example.
+
+```typescript title=index.ts
+export const libraryFunction = (param: number[], reverse: boolean) => {
+  let reversed: number[] = [];
+
+  if (reverse) {
+    param.forEach((n) => table.insert(reversed, 1, n));
+  } else {
+    reversed = param;
+  }
+
+  return table.concat(reversed, ",");
+};
+```
+
+So our function takes an array of numbers like “[1,2,3,4]” and returns a string like “1,2,3,4” or “4,3,2,1”.
+
+This is where unit testing comes in, say a piece of code depends on this function working like it does, and for it to keep working exactly like it does right now, obviously you could use the tried and true method of doing the testing yourself, but unit testing provides an alternative to the manual work.
+
+So, unit testing, where do we begin?
+
+To start off we’re going to need these tools installed:
+
+https://luarocks.org/
+&
+https://olivinelabs.com/busted/
+
+_Luarocks is known to be hard to impossible to install on Windows systems sucessfully, if that’s the case for you, definitely keep reading on, we’re gonna explore a (free) option to do away with running tests locally entirely._
+
+Now that we have busted installed and the `busted` command runs in your terminal of choice successfully we can proceed.
+
+Let’s quickly install busted typings like so `npm install -d busted-tstl`
+
+And we need to add busted-tstl types to our tsconfig.json, you should already have a section like this:
+
+```json title=tsconfig.json
+"types": [
+  "typescript-to-lua/language-extensions",
+  "lua-types/jit",
+]
+```
+
+Simply add `"busted-tstl"` :
+
+```json title=tsconfig.json
+"types": [
+  "typescript-to-lua/language-extensions",
+  "lua-types/jit",
+  "busted-tstl"
+]
+```
+
+Now let’s set up a folder for our tests, by default busted takes a folder named `spec` and runs the files in there, now
+this is personal preference, but usually I name my tests `[filename to be tested]_spec.ts`, the \_spec suffix is once again what busted searches for by default, so we’ll stick to that.
+
+Now our project should look something like this:
+
+![Project setup](/images/testing-project-setup-with-spec-folder.png)
```

Comment:
Use text instead if possible (see above)
If not provide a accurate alt-text description e.g: "File Explorer of vscode showing directory X, File Y... with a red border around File Y..."